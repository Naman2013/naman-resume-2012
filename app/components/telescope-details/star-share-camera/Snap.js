import React from 'react';
import PropTypes from 'prop-types';
import { backgroundImageCover } from '../../../styles/mixins/utilities';

const propTypes = {
  imageURL: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
};

const defaultProps = {
  imageURL: '',
  width: '50px',
  height: '50px',
};

const Snap = ({ imageURL, width, height }) => {
  if (!imageURL) {
    return null;
  }

  return (
    <div
      className="root"
      style={{ backgroundImage: `url(${imageURL})`, width, height }}
    >
      <style jsx>{`
        .root {
          ${backgroundImageCover}
          width: 100%;
          height: 100%;
        }
      `}</style>
    </div>
  );
};

Snap.propTypes = propTypes;
Snap.defaultProps = defaultProps;

export default Snap;
