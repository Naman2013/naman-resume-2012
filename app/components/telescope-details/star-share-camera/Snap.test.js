import React from 'react';
import renderer from 'react-test-renderer';
import Snap from './Snap';

it('should render nothing when no imageURL is provided', () => {
  const tree = renderer.create(<Snap />);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render a container when an image url is provided', () => {
  const tree = renderer.create(<Snap imageURL="foo.jpg" />);
  expect(tree.toJSON()).toMatchSnapshot();
});
