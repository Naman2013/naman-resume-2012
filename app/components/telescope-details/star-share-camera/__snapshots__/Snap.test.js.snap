// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a container when an image url is provided 1`] = `
<div
  className="root"
  data-jsx={2758523860}
  style={
    Object {
      "backgroundImage": "url(foo.jpg)",
      "height": "50px",
      "width": "50px",
    }
  }
/>
`;

exports[`should render nothing when no imageURL is provided 1`] = `null`;
